### DISCUSSION TITLE
Feature request: View page source easily

### DISCUSSION THREAD

**Discussion by FergusFettes** [quarto-dev/quarto-cli#3056] [Q&A] [answered]
Hi folks, absolutely love Quarto, I will be using it for all my writing from now on.

Would it be possible to add the source for the pages directly to the blogs when created? This would make it super easy to get inspiration-- when you are looking at the quarto docs, you could just see the source of the page directly.

I understand that you can get the source if 'code-tools' is active, or (as on many Quanto docs pages) if they have a link to the source code in eg. GitHub. But an option that adds this to the background of every blog page would be super cool.

This idea is inspired by [Gwern](https://www.gwern.net/About)-- on every page of his site, you can just append '.page' to the page to see the markdown source file-- eg, for that link-- https://www.gwern.net/About.page.

Thanks again for the work, its really perfect in every way, and exactly what I have been searching for!

(Oh, one thing missing I noticed so far-- `#| column: margin-left`! Apart from that, spot on :100:.)

Similar discussion: https://github.com/quarto-dev/quarto-cli/discussions/2432

**Comment by mcanouil** [comment, marked as answer]
I don't see how having to change the page extension makes it easier than clicking on the code tools button 🤔
Am I missing something?

Setting the following[^1] at the document or project level:

```yaml
format:
  html: 
    code-tools:
      source: true
      toggle: false
      caption: none
```

https://user-images.githubusercontent.com/8896044/198127027-da669617-771d-435b-ac98-96b137159954.mov

[^1]: from <https://quarto.org/docs/output-formats/html-code.html#code-tools>.
